codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -185,262 +84,520 @@ Pass the shorter list into A for better performance
+    None
   }
-}
 
-exception UnexpectedRegisterDoesNotExist(id)
+  (
+    {
+      id: target.id,
+      status: {
+        fetchingStateId: None,
+      },
+      fetchedEventQueue: mergeSortedEventList(register.fetchedEventQueue, target.fetchedEventQueue),
+      contractAddressMapping: mergedContractAddressMapping,
+      dynamicContracts: mergedDynamicContracts,
+      latestFetchedBlock,
+    },
+    rest,
+  )
+}
 
 /**
 Updates a given register with new latest block values and new fetched
 events.
 */
-let updateRegister = (
+let addItemsToPartition = (
   register: register,
   ~latestFetchedBlock,
   //Events ordered latest to earliest
   ~reversedNewItems: array<Internal.eventItem>,
 ) => {
-  let firstEventBlockNumber = switch register.firstEventBlockNumber {
-  | Some(n) => Some(n)
-  | None => reversedNewItems->Utils.Array.last->Option.map(v => v.blockNumber)
-  }
   {
     ...register,
+    status: {
+      fetchingStateId: None,
+    },
     latestFetchedBlock,
-    firstEventBlockNumber,
     fetchedEventQueue: Array.concat(reversedNewItems, register.fetchedEventQueue),
   }
 }
 
-let makeDynamicContractRegister = (
-  ~registeringEventBlockNumber,
-  ~registeringEventLogIndex,
-  ~dynamicContractRegistrations: array<TablesStatic.DynamicContractRegistry.t>,
-) => {
-  let id: dynamicContractId = {
-    blockNumber: registeringEventBlockNumber,
-    logIndex: registeringEventLogIndex,
-  }
-
-  let contractAddressMapping =
-    dynamicContractRegistrations
-    ->Array.map(d => (d.contractAddress, (d.contractType :> string)))
-    ->ContractAddressingMap.fromArray
-
-  let dynamicContracts =
-    DynamicContractsMap.empty->DynamicContractsMap.add(
-      id,
-      contractAddressMapping->ContractAddressingMap.getAllAddresses,
-    )
-
-  {
-    id: makeDynamicContractRegisterId(id),
-    latestFetchedBlock: {
-      blockNumber: Pervasives.max(registeringEventBlockNumber - 1, 0),
-      blockTimestamp: 0,
-    },
-    contractAddressMapping,
-    dynamicContracts,
-    fetchedEventQueue: [],
-    firstEventBlockNumber: None,
-  }
-}
-
 /*
  Update fetchState, merge registers and recompute derived values
  */
 let updateInternal = (
   fetchState: t,
-  ~registers=fetchState.registers,
-  ~pendingDynamicContracts=fetchState.pendingDynamicContracts,
+  ~partitions=fetchState.partitions,
+  ~nextPartitionIndex=fetchState.nextPartitionIndex,
   ~isFetchingAtHead=fetchState.isFetchingAtHead,
+  ~batchSize=fetchState.batchSize,
+  ~firstEventBlockNumber=fetchState.firstEventBlockNumber,
 ): t => {
-  let registerByLatestBlock = Js.Dict.empty()
-  let add = register => {
-    let key = register.latestFetchedBlock.blockNumber->Js.Int.toString
-    let mergedRegister = switch registerByLatestBlock->Utils.Dict.dangerouslyGetNonOption(key) {
-    | Some(next) => register->mergeWithNextRegister(~next)
-    | None => register
+  let firstPartition = partitions->Js.Array2.unsafe_get(0)
```

Comment:
Just a note here @DZakh, I've been testing your branch on an indexer that reorgs a lot this morning and a reorg crashes the indexer here. First partition does not exist in the case I ran in to.

Reply:
I fixed it yesterday, could you double check that you have the latest version
https://github.com/enviodev/hyperindex/pull/405/commits/10fdcd9a75626bd261b0ca02f108dad0aa47ff36